src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1

```diff
@@ -196,6 +197,31 @@ param(
     default { throw "7-Zip signalled an unknown error (code $exitCode)" }
   }
 
-  $env:ChocolateyPackageInstallLocation = $destination
-  return $destination
+  if ((test-path variable:url) -AND (@($url).count -gt 0))
+  {
+    # If our unzip resulted in a .tar file that matches our input zip file without it's extensions, unzip it
+    # does not work with archive names that contain a '.' before the extensions
+    $matchingtarname = "$destinationNoRedirection\$((split-path -leaf $url).split('.')[0]).tar"
+    Write-Debug "Checking to see if we have a flie called $matchingtarname"
+    If ((Test-path $matchingtarname) -AND (!$RecursiveCall))

```

Comment:
This doesn't make it a recursive call if it can't continue to process down. Either we should make this a recursive call or just run unzip again on the tar. No need to introduce complexity here in the name of saving a small amount of repetition.